File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/kafka/producers/mapping/MelosysEessiMeldingMapperA011.java
package no.nav.melosys.eessi.kafka.producers.mapping;

import no.nav.melosys.eessi.kafka.producers.model.Periode;
import no.nav.melosys.eessi.models.sed.SED;
import no.nav.melosys.eessi.models.sed.medlemskap.impl.MedlemskapA011;
import no.nav.melosys.eessi.models.sed.medlemskap.impl.SvarAnmodningUnntakBeslutning;

class MelosysEessiMeldingMapperA011 extends SvarAnmodningUnntakEessiMeldingMapper<MedlemskapA011> {

    @Override
    MedlemskapA011 hentMedlemskap(SED sed) {
        return (MedlemskapA011) sed.getMedlemskap();
    }

    @Override
    SvarAnmodningUnntakBeslutning hentBeslutning(MedlemskapA011 medlemskap) {
        return SvarAnmodningUnntakBeslutning.INNVILGELSE;
    }

    @Override
    String hentBegrunnelse(MedlemskapA011 medlemskap) {
        return null;
    }

    @Override
    Periode hentDelvisInnvilgetPeriode(MedlemskapA011 medlemskap) {
        return null;
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/service/sed/mapper/fra_sed/melosys_eessi_melding/MelosysEessiMeldingMapperA011.kt
package no.nav.melosys.eessi.service.sed.mapper.fra_sed.melosys_eessi_melding

import no.nav.melosys.eessi.kafka.producers.model.Periode
import no.nav.melosys.eessi.models.sed.SED
import no.nav.melosys.eessi.models.sed.medlemskap.impl.MedlemskapA011
import no.nav.melosys.eessi.models.sed.medlemskap.impl.SvarAnmodningUnntakBeslutning

internal class MelosysEessiMeldingMapperA011 : SvarAnmodningUnntakEessiMeldingMapper<MedlemskapA011?>() {
    override fun hentMedlemskap(sed: SED?): MedlemskapA011? = sed?.medlemskap as MedlemskapA011?

    override fun hentBeslutning(medlemskap: MedlemskapA011?): SvarAnmodningUnntakBeslutning = SvarAnmodningUnntakBeslutning.INNVILGELSE

    override fun hentBegrunnelse(medlemskap: MedlemskapA011?): String? = null

    override fun hentDelvisInnvilgetPeriode(medlemskap: MedlemskapA011?): Periode? = null
}
